src/passes/OptimizeInstructions.cpp

```diff
@@ -350,6 +350,46 @@ struct OptimizeInstructions
           return inner;
         }
       }
+      {
+        // x <<>> (C & (31 | 63))   ==>   x <<>> C'
+        // x <<>> (y & (31 | 63))   ==>   x <<>> y
+        // where '<<>>':
+        //   '<<', '>>', '>>>'. 'rotl' or 'rotr'
+        Type type = curr->type;
+
+        if (type.isInteger()) {
```

Comment:
The code would be simpler without this extra type check. You could also get rid of the `type` variable by using `c->type` instead.